subtests/docker_cli/start/simple.py

```diff
@@ -52,39 +46,39 @@ def _start_container(self, name):
         subargs.append("'echo STARTED: $(date); while :; do sleep 0.1; done'")
         container = AsyncDockerCmd(self, 'run', subargs)
         container.execute()
-        utils.wait_for(lambda: container.stdout.startswith("STARTED"), 5,
-                       step=0.1)
+        ok = utils.wait_for(lambda: container.stdout.startswith("STARTED"), 9,
+                            step=0.1)
+        self.failif(not ok, "Timed out waiting for STARTED from container.\n"
+                    "Output: '%s'" % container.stdout)
 
     def run_once(self):
         # Execute the start command
         super(simple, self).run_once()
         name = self.sub_stuff['container_name']
-        err_msg = ("Start of the %s container failed, but '%s' message is not "
-                   "in the output:\n%s")
-        # Nonexisting container
-        missing_msg = self.config['missing_msg']
+        # Container does not yet exist; 'start' should fail.
         result = mustfail(DockerCmd(self, "start", [name]).execute(), 1)
-        self.failif(missing_msg not in str(result), err_msg
-                    % ("non-existing", missing_msg, result))
+        self.failif_not_in(self.config['missing_msg'], str(result),
+                           "'docker start <nonexistent container>' failed"
+                           " (as expected), but docker error message did not"
+                           " include expected diagnostic.")
 
-        # Running container
+        # Now run the container. The first "start" here should be a NOP.
         self._start_container(name)
         result = mustpass(DockerCmd(self, "start", [name]).execute())
 
-        # Stopped container
+        # Stop container, then restart it.
         mustpass(DockerCmd(self, "kill", [name]).execute())
+        mustpass(DockerCmd(self, "wait", [name]).execute(), 137)
         result = mustpass(DockerCmd(self, "start", [name]).execute())
 
     def postprocess(self):
         super(simple, self).postprocess()
         name = self.sub_stuff['container_name']
         logs = AsyncDockerCmd(self, "logs", ['-f', name])
         logs.execute()
-        utils.wait_for(lambda: logs.stdout.count("\n") == 2, 5, step=0.1)
-        out = logs.stdout
-        self.failif(out.count("\n") != 2, "The container was executed twice, "
-                    "there should be 2 lines with start dates, but is "
-                    "%s.\nContainer output:\n%s" % (out.count("\n"), out))
+        ok = utils.wait_for(lambda: logs.stdout.count("\n") == 2, 20, step=0.1)
+        self.failif(not ok, "Timed out waiting for second STARTED message.\n"
+                    "Output: '%s'" % logs.stdout)
```

Comment:
`docker events` changed output format between 1.9 and 1.10. There's code to deal with that in the events test, I don't want to replicate it.

On reflection, just how bad is it to run `docker logs` every half-second anyway? I'm going to try that and see how it works.